### DISCUSSION TITLE
Problem to install datastore option

### DISCUSSION THREAD

**Discussion by mabroglia** [pymodbus-dev/pymodbus#1192] [Help] [answered]
Hello

First of all I'm sorry if this is not the right forum to report this problem.
I'm trying to install pymodbus 3.0.2, via "pip" in Python 3.8.10. Does anyone have an idea how to solve the problem?
I'm not experienced in Python but if you give me a way I can research.

Best regards and congratulations by pymodbus.
Marco Broglia
#################################
C:\Users\MBROGLIA>pip install -U pymodbus datastore
Requirement already satisfied: pymodbus in c:\users\mbroglia\appdata\local\programs\python\python38\lib\site-packages (3.0.2)
Collecting datastore
  Using cached datastore-0.3.6.tar.gz (30 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ?─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\MBROGLIA\AppData\Local\Temp\pip-install-wo_nqfn\datastore_e61c66bcd2174b939b5147ad7a79b84b\setup.py", line 21
          print 'something went wrong reading the README.md file.'
                ^
      SyntaxError: Missing parentheses in call to 'print'. Did you mean print('something went wrong reading the README.md file.')?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
?─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

C:\Users\MBROGLIA>
#################################

**Comment by janiversen** [comment, marked as answer]
As you can see in the log pymodbus is installed correctly, but second package datastore have a problem.

to verify please try:
  
    pip install pymodbus

    pip install data store

If you refer to the internal datastore of pymodbus, that is installed automatically as part of the pymodbus installation.

**Reply by mabroglia** [reply]
Hello janiversen

Thank you very much for the support.
I did the test as proposed:
#################################
C:\Users\MBROGLIA>pip install pymodbus
Requirement already satisfied: pymodbus in c:\users\mbroglia\appdata\local\programs\python\python38\lib\site-packages (3
.0.2)

C:\Users\MBROGLIA>**pip install data store**
Collecting data
  Downloading data-0.4.tar.gz (7.0 kB)
  Preparing metadata (setup.py) ... done
Collecting store
  Downloading store-2020.2.10-py3-none-any.whl (9.8 kB)
Collecting six
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting decorator
  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting funcsigs
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl (17 kB)
Collecting pony
  Downloading pony-0.7.16-py3-none-any.whl (313 kB)
     ---------------------------------------- 313.9/313.9 kB 414.3 kB/s eta 0:00:00
Installing collected packages: pony, funcsigs, store, six, decorator, data
  DEPRECATION: data is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject
.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is
to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for data ... done
Successfully installed data-0.4 decorator-5.1.1 funcsigs-1.0.2 pony-0.7.16 six-1.16.0 store-2020.2.10

C:\Users\MBROGLIA>
#################################
The behaviour was different, is not clear to me if 'wheel' package was installed correctly, I'll check if everything is working.

Best Regards janiversen

**Reply by janiversen** [reply]
sounds good, but please be aware I cannot help you with problems around the data_store project, I have no idea what it is. Seem from your log, pymodbus is installed correctly.

**Comment by janiversen** [comment]
Did you solve your problems ?
